tests/RobotFramework/tests/tedge/device_id.robot

```diff
@@ -0,0 +1,69 @@
+*** Settings ***
+Documentation       This suite covers all usecases in the issue #3369
+
+Resource            ../../resources/common.resource
+Library             Cumulocity
+Library             ThinEdgeIO
+Library             ../../.venv/lib/python3.11/site-packages/robot/libraries/String.py
+
+Suite Teardown      Get Logs
+Test Setup          Custom Setup
+
+Test Tags           theme:c8y    theme:cli
+
+
+*** Test Cases ***
+Use explicit device id during cert creation
+    Execute Command    tedge config set device.id ${FOO}
+    ${output}=    Execute Command    tedge cert create --device-id ${BAR}
+    Should Contain    ${output}    CN=${BAR}
+
+    ${output}=    Execute Command    tedge config get device.id    strip=${True}
+    Should Be Equal    ${output}    ${BAR}
+
+    ${output}=    Execute Command    tedge connect c8y    ignore_exit_code=${True}
+    Should Contain    ${output}    device id: ${BAR}
+
+Use default device.id
+    Execute Command    tedge config set device.id ${FOO}
+    ${output}=    Execute Command    tedge cert create
+    Should Contain    ${output}    CN=${FOO}
+
+    ${output}=    Execute Command    tedge config get device.id    strip=${True}
+    Should Be Equal    ${output}    ${FOO}
+
+    ${output}=    Execute Command    tedge connect c8y    ignore_exit_code=${True}
+    Should Contain    ${output}    device id: ${FOO}
+
+Use default device.id 2
```

Comment:
Can you add a more descriptive test name here? It is too similar to `Use default device.id` which makes it unclear what the test case is about.

Reply:
Renamed to `Use device id from cert`